linkerd/identity/src/metrics.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{Credentials, DerX509};
+use linkerd_error::Result;
+use linkerd_metrics::prom;
+use std::{
+    sync::atomic::AtomicU64,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct CertMetrics {
+    refresh_ts: prom::Gauge<f64, AtomicU64>,
+    expiry_ts: prom::Gauge<f64, AtomicU64>,
+    refreshes: prom::Counter,
+}
+
+/// Implements `Credentials`, recording metrics about certificate updates.
+pub struct WithCertMetrics<C> {
+    inner: C,
+    metrics: CertMetrics,
+}
+
+impl CertMetrics {
+    pub fn register(registry: &mut prom::Registry) -> Self {
+        let expiry_ts = prom::Gauge::default();
+        registry.register_with_unit(
+            "expiration_timestamp",
+            "Time when the this proxy's current mTLS identity certificate will expire (in seconds since the UNIX epoch)",
+           prom::Unit::Seconds, expiry_ts.clone()
+        );
+
+        let refresh_ts = prom::Gauge::default();
+        registry.register_with_unit(
+            "refresh_timestamp",
+            "Time when the this proxy's current mTLS identity certificate were last updated",
+            prom::Unit::Seconds,
+            refresh_ts.clone(),
+        );
+
+        let refreshes = prom::Counter::default();
+        registry.register(
+            "refreshes",
+            "The total number of times this proxy's mTLS identity certificate has been refreshed by the Identity provider",
+            refreshes.clone(),
+        );
```

Comment:
As it is, this will get udpated only on successful refresh. I wonder whether we want to capture situations when the `Credentials::set_certificate` actually fails. 

For example we can have a label on this metrics that indicates the outcome of an attempted refresh and always set it whenever certs have been received from the provider